### PR TITLE
[Shezad Sulyman Imran Hassan] iP

### PR DESCRIPTION
**Pull request by SHEZADHASSAN22**
# DukePro

*DukePro* frees your mind of having to remember things you need to do. It's,
* test based
* easy to learn
* ~~fast~~ Super Fast to use

All you need to do it,
1. download it from [here](http://github.com)
2. double click it.
3. add your tasks,
4. let it manage your tasks for you!

And it is **Free**

Here is the `main` method:
```java
public static void main(String[] args) {
        new Ui().greeting();
        new Duke("dukeData.txt").run();
    }
```

Features:
- [x] Managing tasks
- [ ] Managing Deadlines (coming soon) 💯 

As Kanye West said:
> We're living the future so
> the present is our past.

### CODE REVIEW

**Comment by darrenhoon** [comment]
Your header **DukePro** looks great!

```
new Ui().greeting();
```
hey, I like how clean this bit of code is 👍

Regarding "Kanye West", perhaps it should be "Kanye _East_"?

File: src/main/java/duke/Event.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package duke;
+
+import java.time.LocalDateTime;
+
+public class Event extends Task{
```
Comment (darrenhoon) [inline src/main/java/duke/Event.java]: Perhaps there could be a header comment here to help users understand what this code does?

```
/**
 * The Event class is a child class of Task.
 * Contains additional "date" string parameter.
 * @author  
 * @version 
 * @since  
 */
```

Similar comment for the other classes to ensure you don't lose marks unnecessarily due to CheckStyle issues :)

File: src/main/java/duke/TaskList.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        } catch (StringIndexOutOfBoundsException e) {
+            return("OOPS!!! You cannot mark nothing as done!");
+        } catch (NumberFormatException e) {
+            return("OOPS!!! Must be a number bodoh");
+        } catch (IndexOutOfBoundsException e) {
+            return("OOPS!!! NUmber doesnt exist");
+        }
+    }
+
+    public List<Task> getTasks() {
```
Comment (darrenhoon) [inline src/main/java/duke/TaskList.java:168]: missing header comment here for method

File: text-ui-test/EXPECTED.TXT [hunk 3]
```diff
@@ -5,3 +5,39 @@ Hello from
 | |_| | |_| |   <  __/
 |____/ \__,_|_|\_\___|
 
+What can I do for you?
```
Comment (darrenhoon) [inline text-ui-test/EXPECTED.TXT:8]: Great to see that you are doing tests! Your test-driven development greatly helps to reduce the number of bugs that you might run into in the future, which can save time. Great job!

File: src/main/java/duke/TaskList.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        return this.tasks;
+    }
+
+    /**
+     * Prints the list of todos sequentially.
+     *
```
Comment (darrenhoon) [inline src/main/java/duke/TaskList.java:127]: Perhaps the header comments should not have empty lines based on JavaDocs convention?

**Review by darrenhoon — COMMENTED** [review]
Overall, the structure looks good as it helps to ensure modularity, such as Event, Todo, and Deadline inheriting from Task.

Testing shows great effort from you to ensure your code's rigidity against fringe cases while meaningful names ensure that anyone who wants to join your project can easily follow along to understanding the code and quickly improve the onboarding process.

File: src/main/java/duke/Event.java (hunk 1 above)
Comment (SHEZADHASSAN22) [reply src/main/java/duke/Event.java]: Cool thanks bro.

File: src/main/java/duke/Task.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    public Task(String task, String taskSymbol) {
+        this.task = task;
+        this.isDone = false;
+        this.taskSymbol = taskSymbol;
+    }
+
+    /**
+     * Represents task in the from that it is stored in the data file.
+     *
+     * @return String of the task in the form required by teh data file.
```
Comment (CheahYan) [inline src/main/java/duke/Task.java:45]: I think there is a typo for your comment here. for "teh"

File: src/main/java/duke/Parser.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            toReply = taskList.markTaskDone(userInput);
+            storage.updateDataSet(taskList.getTasks());
+        } else if (userInput.toLowerCase().startsWith("delete")) {
+            toReply = taskList.deleteTask(userInput);
+            storage.updateDataSet(taskList.getTasks());
+        } else if (userInput.toLowerCase().startsWith("find")) {
+            toReply = taskList.findTask(taskList.getTasks(), userInput);
+        } else {
+            toReply = taskList.addTask(userInput, storage.getDataPath());
+        }
```
Comment (CheahYan) [inline src/main/java/duke/Parser.java:40]: Do you think this may be too wordy for one method? Perhaps having some methods to abstract out some  code will make it more readable?

**Review by CheahYan — COMMENTED** [review]
Overall, I find your code very easy to read and it is clean for the most part. It also follows the coding standards very strictly and I was not able to spot many mistakes. LGTM!

File: src/main/java/duke/Storage.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
```
Comment (CheahYan) [inline src/main/java/duke/Storage.java:12]: Missing header comment here for your class!

File: src/main/java/duke/Task.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.time.LocalDateTime;
+
+
+public class Task {
```
Comment (CheahYan) [inline src/main/java/duke/Task.java:6]: Missing header comments here too for your class.

**Review by CheahYan — COMMENTED** [review]
Updates to my previous peer review. Realised that I did not looked at Javadoc-related issues. There are missing header comments for some of your classes. Otherwise, the code follows the coding standards set by the module. Good job! LGTM!
